figma-shop/src/app/pages/products/products.component.ts

```diff
@@ -0,0 +1,32 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { map } from 'rxjs/operators';
+import { IProduct } from 'src/app/shared/interfaces/product.interface';
+import { ProductServiceService } from '../../shared/services/product-service.service';
+
+
+@Component({
+  selector: 'app-products',
+  templateUrl: './products.component.html',
+  styleUrls: ['./products.component.scss']
+})
+export class ProductsComponent implements OnInit {
+  public allProducts: Array<IProduct>;
```

Comment:
For your future projects, writing `I` before interfaces that's an outdated practice... 
At our projects we always write as `ProductInterface` or `ProductModel`, but in your project this is up to you, of course.